psyneulink/core/llvm/codegen.py

```diff
@@ -28,16 +28,28 @@ def __init__(self, ctx, builder, func_globals, func_params, arg_in, arg_out):
         self.func_params = func_params
         self.arg_in = arg_in
         self.arg_out = arg_out
-        self.register = {}
 
         #setup default functions
-        def _vec_sum(x):
-            dim = len(x.type.pointee)
-            output_scalar = builder.alloca(ctx.float_ty)
-            # Get the pointer to the first element of the array to convert from [? x double]* -> double*
-            vec_u = builder.gep(x, [ctx.int32_ty(0), ctx.int32_ty(0)])
-            builder.call(ctx.import_llvm_function("__pnl_builtin_vec_sum"), [vec_u, ctx.int32_ty(dim), output_scalar])
-            return output_scalar
+        def _list_sum(x):
+            # HACK: Obtain generic addition function by visiting the add node
+            # this should ideally be moved to an explicit helper
+            add_func = self.visit_Add(None)
```

Comment:
The comment is not quite clear here. Does the `visit_Add` here return the right addition for different element types (float/int/array)?

Reply:
yes - this is the polymorphic addition function as used when visiting an addition node

Reply:
yes, but you pass `None`, so what's the mechanism to infer the correct type? is that taken from the parser state?